include/toolchain/common.h

```diff
@@ -26,6 +26,10 @@
 #define TASK_ENTRY_CPP  extern "C"
 #endif
 
+#ifndef ZRESTRICT
+#define ZRESTRICT restrict
+#endif
```

Comment:
This is not correct when compiling C++ code (i.e. when `__cplusplus` is defined).

If toolchain-specific C++ restrict qualifier is not defined, the default behaviour should be to use `restrict` keyword for C and ignore restrict qualification for C++ as the C++ standard does not support such a qualifier.

```suggestion
#ifndef ZRESTRICT
#ifndef __cplusplus
#define ZRESTRICT restrict
#else
#define ZRESTRICT
#endif
#endif /* !defined(ZRESTRICT) */
```